qiskit_experiments/calibration_management/backend_calibrations.py

```diff
@@ -196,7 +213,149 @@ def export_backend(self) -> Backend:
 
         backend.defaults().qubit_freq_est = self.get_qubit_frequencies()
         backend.defaults().meas_freq_est = self.get_meas_frequencies()
-
-        # TODO: build the instruction schedule map using the stored calibrations
+        backend.default().instruction_schedule_map = self._inst_map
 
         return backend
+
+    def inst_map_add(
+        self,
+        instruction_name: str,
+        qubits: Tuple[int],
+        schedule_name: Optional[str] = None,
+        assign_params: Optional[Dict[Union[str, ParameterKey], ParameterValueType]] = None,
+    ):
+        """Update a single instruction in the instruction schedule map.
+
+        This method can be used to update a single instruction for the given qubits but
+        it can also be used by experiments that define custom gates with parameters
+        such as the :class:`Rabi` experiment. In a Rabi experiment there is a gate named
+        "Rabi" that scans a pulse with a custom amplitude. Therefore we would do
+
+        .. code-block:: python
+
+            cals.inst_map_add("Rabi", (0, ), "xp", assign_params={"amp": Parameter("amp")})
+
+        to temporarily add a pulse for the Rabi gate in the instruction schedule map. This
+        then allows calling :code:`transpile(circ, inst_map=cals.instruction_schedule_map)`.
+
+        Args:
+            instruction_name: The name of the instruction to add to the instruction schedule map.
+            qubits: The qubits to which the instruction will apply.
+            schedule_name: The name of the schedule. If None is given then we assume that the
+                schedule and the instruction have the same name.
+            assign_params: An optional dict of parameter mappings to apply. See for instance
+                :meth:`get_schedule` of :class:`Calibrations`.
+        """
+        schedule_name = schedule_name or instruction_name
+
+        inst_map_args = None
+        if assign_params is not None:
+            inst_map_args = assign_params.keys()
+
+        self._inst_map.add(
+            instruction=instruction_name,
+            qubits=qubits,
+            schedule=self.get_schedule(schedule_name, qubits, assign_params),
+            arguments=inst_map_args,
+        )
+
+    def inst_map_remove(self, instruction: str, qubits: Tuple[int]):
+        """Remove a single instruction from the inst_map."""
+        self._inst_map.remove(instruction, qubits)
```

Comment:
This should be managed by instmap. We should keep the cal class as simple as possible.

Reply:
Good point. We can do this via the property. See d6fe8de